### PR TITLE
Login Credentials extraction

### PR DESCRIPTION
**Pull request by wizzard**
* add login credential extraction for 3 types: header, url, body JSON
* login credential information is loaded from a local JSON file (`px_login_creds_settings_filename`)
* rework Dockerfile
* rework nginx.conf

### CODE REVIEW

**Review by shaul-badusa-px — APPROVED** [review]

**Comment by shaul-badusa-px** [comment]
@pxjohnny Do you know why the ci is failing?

File: lib/px/utils/pxtimer.lua [hunk 1]
```diff
@@ -13,6 +13,7 @@ function M.application(px_configutraion_table)
 	local px_logger = require ("px.utils.pxlogger").load(px_config)
 	local px_constants = require("px.utils.pxconstants")
 	local buffer = require "px.utils.pxbuffer"
+	local px_creds = require ("px.utils.pxlogin_credentials").load(px_config)
```
Comment (pxjohnny) [inline lib/px/utils/pxtimer.lua:16]: why is this needed here?

**Comment by pxjohnny** [comment]
seems to be related to the updated Docker: Can't locate Test/Nginx/Socket/Lua.pm in @INC (you may need to install the Test::Nginx::Socket::Lua module). probably just missing a dependency for testing.

**Review by pxjohnny — APPROVED** [review]
Please make sure you manual test this change
